### DISCUSSION TITLE
DevOps success for business leaders: Making business sense

### DISCUSSION THREAD

**Discussion by ayodejiayodele** [community/community#33143] [Discover: GitHub Best Practices]
With everything that has happened over the last few years, business decision-making has become more critical, especially when it comes to allocating and prioritising budgets. You can no longer justify throwing money on fancy IT projects. While some businesses barely survived, some businesses thrived through sheer resilience and nimbleness.

<a id="7-reasons-why-DevOps-is-serious-business"></a>
Here are `7 reasons why` adopting DevOps is critical to the survival of your business for the next decade.

If you would like to know what DevOps means, [first read this paragraph](#what-is-devops). 

### Tl;dr

1. [Software delivery is the new normal](#1-software-delivery-is-the-new-normal)
2. [Disruptors are here to stay](#2-Disruptors-are-here-to-stay)
3. [Favorable for shareholders](#3-Favorable-for-shareholders)
4. [Retain talent](#4-Retain-talent)
5. [Massive cost savings](#5-Massive-cost-savings)
6. [Protect the planet](#6-Protect-the-planet)
7. [Stay secure and stay prepared](#7-Stay-secure-and-stay-prepared)

<a id="1-software-delivery-is-the-new-normal"></a>
### 1. Software delivery is the new normal
Every company is becoming a software company. Software is now the market differentiator that draws customers and increases market share. Customers are increasingly demanding speed, efficiency, and lower prices. You can only meet this need with technology, and software drives technology. 

<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/63050478/187430971-b3ceb6a1-74fa-4ba4-8310-a1ecf676ea46.png">
  <source media="(prefers-color-scheme: light)" srcset="https://user-images.githubusercontent.com/63050478/187325850-f07d0441-2b6e-4866-a69b-a884a7bc64df.png">
  <img alt="Software is eating our world." src="https://user-images.githubusercontent.com/63050478/187325850-f07d0441-2b6e-4866-a69b-a884a7bc64df.png">
</picture>

Virtually, everything around us is driven by software, or “firmware” which I call software’s little brother. 🤓 Take a look around you, your smartphone and the apps on it, the TV, your food source and its delivery, [trains](https://en.wikipedia.org/wiki/Automatic_train_operation), [X-ray equipment](https://www.itnonline.com/article/takeoff-iot-radiology), [electric toothbrush](https://www.nature.com/articles/s41415-019-0551-9), all are driven by software and firmware-powered integrated circuits (IC). If you have software development functions on the fringes, you should consider making it central/core to your business.
With DevOps, you are able to incrementally build software and manage the pace of customer demands in an efficient, sustainable, and secure manner. DevOps allows you to think big but start small.
<a id="2-disruptors-are-here-to-stay"></a>
### 2. Disruptors are here to stay
Technology disruptors are emerging in every sector. What market technology trends are out there that could disrupt your business model? How quickly can you respond if disruptions occur? What is the competition doing? Do you have the capabilities to pursue new products and services? One way to prepare for this is to self-disrupt your business. Self-disruption involves building and delivering products that compete in the same market as another of your products.

<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/63050478/187431580-073ae9a2-a5b9-448e-8675-ce60f7f6c8ce.png">
  <source media="(prefers-color-scheme: light)" srcset="https://user-images.githubusercontent.com/63050478/187326071-9da81091-6c44-4420-9890-e21ad866bcd6.png">
  <img alt="If you don't cannibalise yourself, someone else will." src="https://user-images.githubusercontent.com/63050478/187326071-9da81091-6c44-4420-9890-e21ad866bcd6.png">
</picture>

For example, Apple launched the iPhone to compete in the same market as the iPod, a cash cow that was responsible for up to 50% of Apple’s revenue, back in 2007. Today, the iPod has been discontinued. Imagine if Apple stayed comfortable with the iPod and another inventor emerged with the iPhone. ⁉️ DevOps encourages experimentation, continuous learning, and the unfaltering delivery of multiple software products and versions in concurrent environments. It enables business agility and a means for organisations to innovate and adapt quickly in an increasingly changing world. `Self-disruption is self-preservation`.
<a id="3-Favorable-for-shareholders"></a>
### 3. Favorable for shareholders
Organisations with elite DevOps practices (teams excelling in the [four key metrics](https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance) of software delivery performance) tend to outperform their DevOps-less counterparts in profitability. Most of these companies see 60% higher revenue and profit growth than their counterparts and are [2.4 times more likely to grow by over 20%](https://docs.broadcom.com/doc/how-agile-and-devops-enable-digital-readiness-and-transformation). With DevOps, you are able to build more superior products through collective collaboration of your teams, releasing to market early, continually receiving product feedback, and creating more reliable systems. This undoubtedly wins new customers and retains existing customers. In other words, it increases market share.
<a id="4-Retain-talent"></a>
### 4. Retain talent
Work is no longer as we know it. Workers have more flexible arrangements and more options in choosing employers that align with their values and work-life balance needs. DevOps has been proven to boost productivity and reduce burnout due to the automation and collaboration it enables. This allows IT employees to add more value to the work they do and live well. As an employer, a DevOps culture can attract and keep top talents, making you sought after in the job market.
<a id="5-Massive-cost-savings"></a>
### 5. Massive cost savings 
DevOps is good for the bottom line, you can do more with less. With DevOps, you can dynamically scale your infrastructure up and down to adjust to your customer growth and resource demands. You don’t have to invest a huge upfront cost on system infrastructure, cooling, or power. You also save on manual processes and inefficiencies within your value stream and software supply chain.
<a id="6-Protect-the-planet"></a>
### 6. Protect the planet
As DevOps is a [Lean](https://en.wikipedia.org/wiki/DevOps#Relationship_to_other_approaches) software engineering practice, it encourages you to minimise waste. Therefore, you are not running more resources than you need to, thereby consuming less power and energy. You are also able to control and measure your carbon footprint. Your software ecosystem emits carbon too, not just cows! This topic is increasingly becoming a make-or-break factor for investors and employees in choosing a company. Makes sense! 😼

<a id="7-Stay-secure-and-stay-prepared"></a>
### 7. Stay secure and stay prepared
Gone are the days when cyberattacks are only state-sponsored or targeted. Every technology leader is challenged with enabling the security and reliability of their software products. With DevOps, more specifically DevSecOps (i.e. DevOps wearing a security hat ⛑), you are able to analyse and monitor software code as well as the production system in order to better anticipate and mitigate problems and breaches.

<a id="what-is-devops"></a>
## What is DevOps?
DevOps is the modern way of doing IT. It combines people, processes, and tools in a way that is fine-tuned to produce efficiency, foster collaboration of teams, and automate the delivery of software products to customers and end users in a reliable, secure, efficient, and continuous manner. It is a radical cultural shift from traditional IT in the software development lifecycle. DevOps is grouped into capabilities, and these capabilities include 40+ practices varying from usage of technology, change of business processes, and team structure. These practices and capabilities can be adopted both incrementally and independently.

### The infinity loop
DevOps is characterised by the `infinity loop` ♾️, which means it is ever-evolving and non-stop. This one thing is common to many DevOps illustrations. They might have different stages or different colours, but they tend to always have the infinity shape. What makes DevOps brilliant is its intrinsic evolution, you are always tinkering with better ways to do things to obtain better outcomes. You (and your software) are guaranteed to be better than yesterday.

![devops](https://user-images.githubusercontent.com/63050478/187428045-6b78dfa1-1123-4128-a1a0-0460384f5a46.png)
_DevOps has many faces, but `infinite` is common to them._

Now let’s get to the [7 reasons why](#7-reasons-why-DevOps-is-serious-business) investing in DevOps makes business sense.

I hope this post has provided some food for thought about why your organisation needs DevOps to excel. GitHub provides DevOps tools and guides that can help. To learn more, check out our [DevOps Resources Page](https://resources.github.com/devops/). 

## Useful Links
1. [Getting started with DevOps - GitHub](https://resources.github.com/devops/)
1. [Why software is eating the world - Wall Street Journal](https://www.wsj.com/articles/SB10001424053111903480904576512250915629460)
1. Measure your carbon footprint through your cloud provider: [Azure](https://azure.microsoft.com/en-au/blog/microsoft-sustainability-calculator-helps-enterprises-analyze-the-carbon-emissions-of-their-it-infrastructure/), [AWS](https://aws.amazon.com/aws-cost-management/aws-customer-carbon-footprint-tool/), [GCP](https://cloud.google.com/carbon-footprint)
1. [Digital workplace trends you can't ignore - Gartner](https://www.gartner.com/smarterwithgartner/digital-workplace-trends-you-cant-ignore)
2. [Learn how to make software development more sustainable](https://docs.microsoft.com/en-us/learn/modules/sustainable-software-engineering-overview/)

**Comment by nursopian** [comment]
https://support.github.com/ticket/1796792

**Reply by ZoelefackReynaNk** [reply]
Cool

**Reply by Ryzz123** [reply]
Nice

**Comment by Aii56** [comment]
I want to be a successful trader

**Comment by Aii56** [comment]
I want to be successful

**Comment by judeVector** [comment]
Wow this is very nice writeup

**Reply by ayodejiayodele** [reply]
Thanks @CodeJude 👍 Watch out for the next one 🌟

**Reply by devnullone** [reply]
Awesome article!

**Comment by ericbarrionuevo** [comment]
of course, i like the information of devope related to encourages experimentation, continuous learning.

**Comment by arpan-mondal** [comment]
Very Nicely Explained. It was a good read.

**Reply by kiritodeveloper** [reply]
Hello

**Reply by ayodejiayodele** [reply]
Thanks for the feedback @arpan-mondal, I’m glad you enjoyed reading it 👍🏾

**Comment by ReLBk** [comment]
I appreciate your explanatory article,when I look at dev ops I am humbled and proud to be a part of the process and community, lot's of people might not see the value, but I certainly do.

**Comment by ethanlevy007** [comment]
looks cool[,](https://www.laddboxshoppen.se/) I have a question. Can I use it for my teaching staff in IT purposes?

**Reply by ayodejiayodele** [reply]
@ethanlevy007 Absolutely! Please feel free to use it to teach others. I’d be happy to share more learning resources that also helped me along the way. 🙂📚

**Comment by ReLBk** [comment]
It would depend on your lesson plan, also I don't know if that would qualify as a commercial use,my assumption or conclusion based on what I know is that, assuming you're not using to resale without applying to an agreement with provider than,the answer would really depend on your use case's, generally speaking anyone can download open source, but it's not legal to sell or reproduce without expressed permission and accompanied license and agreement  with arbitration, disclaimer and license/privacy policy displayed somewhere in the beginning of program/1st page,introduction ect.

**Comment by hudsontrius80A** [comment]
> With everything that has happened over the last few years, business decision-making has become more critical, especially when it comes to allocating and prioritising budgets. You can no longer justify throwing money on fancy IT projects. While some businesses barely survived, some businesses thrived through sheer resilience and nimbleness.
> 
> <a id="7-reasons-why-DevOps-is-serious-business"></a>
> Here are `7 reasons why` adopting DevOps is critical to the survival of your business for the next decade.
> 
> If you would like to know what DevOps means, [first read this paragraph](#what-is-devops). 
> 
> ## 7 reasons why DevOps is serious business
> ### Tl;dr
> 
> 1. [Software delivery is the new normal](#1-software-delivery-is-the-new-normal)
> 2. [Disruptors are here to stay](#2-Disruptors-are-here-to-stay)
> 3. [Favorable for shareholders](#3-Favorable-for-shareholders)
> 4. [Retain talent](#4-Retain-talent)
> 5. [Massive cost savings](#5-Massive-cost-savings)
> 6. [Protect the planet](#6-Protect-the-planet)
> 7. [Stay secure and stay prepared](#7-Stay-secure-and-stay-prepared)
> 
> <a id="1-software-delivery-is-the-new-normal"></a>
> ### 1. Software delivery is the new normal
> Every company is becoming a software company. Software is now the market differentiator that draws customers and increases market share. Customers are increasingly demanding speed, efficiency, and lower prices. You can only meet this need with technology, and software drives technology. 
> 
> <picture>
>   <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/63050478/187430971-b3ceb6a1-74fa-4ba4-8310-a1ecf676ea46.png">
>   <source media="(prefers-color-scheme: light)" srcset="https://user-images.githubusercontent.com/63050478/187325850-f07d0441-2b6e-4866-a69b-a884a7bc64df.png">
>   <img alt="Software is eating our world." src="https://user-images.githubusercontent.com/63050478/187325850-f07d0441-2b6e-4866-a69b-a884a7bc64df.png">
> </picture>
> 
> 
> Virtually, everything around us is driven by software, or “firmware” which I call software’s little brother. 🤓 Take a look around you, your smartphone and the apps on it, the TV, your food source and its delivery, [trains](https://en.wikipedia.org/wiki/Automatic_train_operation), [X-ray equipment](https://www.itnonline.com/article/takeoff-iot-radiology), [electric toothbrush](https://www.nature.com/articles/s41415-019-0551-9), all are driven by software and firmware-powered integrated circuits (IC). If you have software development functions on the fringes, you should consider making it central/core to your business.
> With DevOps, you are able to incrementally build software and manage the pace of customer demands in an efficient, sustainable, and secure manner. DevOps allows you to think big but start small.
> <a id="2-disruptors-are-here-to-stay"></a>
> ### 2. Disruptors are here to stay
> Technology disruptors are emerging in every sector. What market technology trends are out there that could disrupt your business model? How quickly can you respond if disruptions occur? What is the competition doing? Do you have the capabilities to pursue new products and services? One way to prepare for this is to self-disrupt your business. Self-disruption involves building and delivering products that compete in the same market as another of your products.
> 
> <picture>
>   <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/63050478/187431580-073ae9a2-a5b9-448e-8675-ce60f7f6c8ce.png">
>   <source media="(prefers-color-scheme: light)" srcset="https://user-images.githubusercontent.com/63050478/187326071-9da81091-6c44-4420-9890-e21ad866bcd6.png">
>   <img alt="If you don't cannibalise yourself, someone else will." src="https://user-images.githubusercontent.com/63050478/187326071-9da81091-6c44-4420-9890-e21ad866bcd6.png">
> </picture>
> 
> For example, Apple launched the iPhone to compete in the same market as the iPod, a cash cow that was responsible for up to 50% of Apple’s revenue, back in 2007. Today, the iPod has been discontinued. Imagine if Apple stayed comfortable with the iPod and another inventor emerged with the iPhone. ⁉️ DevOps encourages experimentation, continuous learning, and the unfaltering delivery of multiple software products and versions in concurrent environments. It enables business agility and a means for organisations to innovate and adapt quickly in an increasingly changing world. `Self-disruption is self-preservation`.
> <a id="3-Favorable-for-shareholders"></a>
> ### 3. Favorable for shareholders
> Organisations with elite DevOps practices (teams excelling in the [four key metrics](https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance) of software delivery performance) tend to outperform their DevOps-less counterparts in profitability. Most of these companies see 60% higher revenue and profit growth than their counterparts and are [2.4 times more likely to grow by over 20%](https://docs.broadcom.com/doc/how-agile-and-devops-enable-digital-readiness-and-transformation). With DevOps, you are able to build more superior products through collective collaboration of your teams, releasing to market early, continually receiving product feedback, and creating more reliable systems. This undoubtedly wins new customers and retains existing customers. In other words, it increases market share.
> <a id="4-Retain-talent"></a>
> ### 4. Retain talent
> Work is no longer as we know it. Workers have more flexible arrangements and more options in choosing employers that align with their values and work-life balance needs. DevOps has been proven to boost productivity and reduce burnout due to the automation and collaboration it enables. This allows IT employees to add more value to the work they do and live well. As an employer, a DevOps culture can attract and keep top talents, making you sought after in the job market.
> <a id="5-Massive-cost-savings"></a>
> ### 5. Massive cost savings 
> DevOps is good for the bottom line, you can do more with less. With DevOps, you can dynamically scale your infrastructure up and down to adjust to your customer growth and resource demands. You don’t have to invest a huge upfront cost on system infrastructure, cooling, or power. You also save on manual processes and inefficiencies within your value stream and software supply chain.
> <a id="6-Protect-the-planet"></a>
> ### 6. Protect the planet
> As DevOps is a [Lean](https://en.wikipedia.org/wiki/DevOps#Relationship_to_other_approaches) software engineering practice, it encourages you to minimise waste. Therefore, you are not running more resources than you need to, thereby consuming less power and energy. You are also able to control and measure your carbon footprint. Your software ecosystem emits carbon too, not just cows! This topic is increasingly becoming a make-or-break factor for investors and employees in choosing a company. Makes sense! 😼
> 
> <a id="7-Stay-secure-and-stay-prepared"></a>
> ### 7. Stay secure and stay prepared
> Gone are the days when cyberattacks are only state-sponsored or targeted. Every technology leader is challenged with enabling the security and reliability of their software products. With DevOps, more specifically DevSecOps (i.e. DevOps wearing a security hat ⛑), you are able to analyse and monitor software code as well as the production system in order to better anticipate and mitigate problems and breaches.
> 
> <a id="what-is-devops"></a>
> ## What is DevOps?
> DevOps is the modern way of doing IT. It combines people, processes, and tools in a way that is fine-tuned to produce efficiency, foster collaboration of teams, and automate the delivery of software products to customers and end users in a reliable, secure, efficient, and continuous manner. It is a radical cultural shift from traditional IT in the software development lifecycle. DevOps is grouped into capabilities, and these capabilities include 40+ practices varying from usage of technology, change of business processes, and team structure. These practices and capabilities can be adopted both incrementally and independently.
> 
> ### The infinity loop
> DevOps is characterised by the `infinity loop` ♾️, which means it is ever-evolving and non-stop. This one thing is common to many DevOps illustrations. They might have different stages or different colours, but they tend to always have the infinity shape. What makes DevOps brilliant is its intrinsic evolution, you are always tinkering with better ways to do things to obtain better outcomes. You (and your software) are guaranteed to be better than yesterday.
> 
> ![devops](https://user-images.githubusercontent.com/63050478/187428045-6b78dfa1-1123-4128-a1a0-0460384f5a46.png)
> _DevOps has many faces, but `infinite` is common to them._
> 
> 
> 
> Now let’s get to the [7 reasons why](#7-reasons-why-DevOps-is-serious-business) investing in DevOps makes business sense.
> 
> I hope this post has provided some food for thought about why your organisation needs DevOps to excel. GitHub provides DevOps tools and guides that can help. To learn more, check out our [DevOps Resources Page](https://resources.github.com/devops/). 
> 
> ## Useful Links
> 1. [Getting started with DevOps - GitHub](https://resources.github.com/devops/)
> 1. [Why software is eating the world - Wall Street Journal](https://www.wsj.com/articles/SB10001424053111903480904576512250915629460)
> 1. Measure your carbon footprint through your cloud provider: [Azure](https://azure.microsoft.com/en-au/blog/microsoft-sustainability-calculator-helps-enterprises-analyze-the-carbon-emissions-of-their-it-infrastructure/), [AWS](https://aws.amazon.com/aws-cost-management/aws-customer-carbon-footprint-tool/), [GCP](https://cloud.google.com/carbon-footprint)
> 1. [Digital workplace trends you can't ignore - Gartner](https://www.gartner.com/smarterwithgartner/digital-workplace-trends-you-cant-ignore)
> 2. [Learn how to make software development more sustainable](https://docs.microsoft.com/en-us/learn/modules/sustainable-software-engineering-overview/)
> 

Thanks

**Comment by Mar9667** [comment]
happy  writeup !

**Comment by GeauxWeisbeck4** [comment]
Great article! Well, great content in general with the length, flow and everything.

**Reply by ayodejiayodele** [reply]
Thanks, more to come 🥂

**Comment by Aii56** [comment]
How you doing
